### ISSUE TITLE
NoResults view controller VoiceOver: suggested improvements for root view, multiline descriptions, and first element read

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/WordPress-iOS#11487]
Steps to reproduce:

1. Go to Settings > General > Accessibility > VoiceOver and toggle the setting on.
1. Go to WPiOS > My Sites > [any site] > Blog Posts > Search, enter text until you see no results.
1. Are the contents of the screen announced properly? Try accessing the things on the screen.
1. Go to Trashed. Repeat step 3.
1. Go to Scheduled. Repeat step 3.
1. Go to Reader > Saved Posts. Repeat step 3. (Note the bookmark icon is not read aloud.)
1. Go to Notifications > Comments and Likes. Repeat step 3.
1. Go to Reader > My Likes. Repeat step 3.

Result: when using VoiceOver, screens with no results are difficult to work with. ([2m2s](https://cloudup.com/ciWHIWb6FXI))

Note: testing this is easier when using a newer site without very much content.

<img src="https://user-images.githubusercontent.com/1119271/56323176-4d00e080-6128-11e9-8216-b0a7063cc56a.png" width="200" alt="noresults-blog-posts-search"> <img src="https://user-images.githubusercontent.com/1119271/56323174-4d00e080-6128-11e9-80f4-3f475f534d74.png" width="200" alt="noresults-reader-saved-posts"> <img src="https://user-images.githubusercontent.com/1119271/56323173-4c684a00-6128-11e9-9106-cd872fe9e59b.png" width="200" alt="noresults-notifications-comments">
<p style="clear:both;"><sup>Tested with WP Internal 12.2.0.20190412 on iPhone 6S iOS 12.2.</sup></p>

Suggested improvements noted by @etoledom:

* The root view should be a whole accessibility element to avoid VoiceOver from reading elements behind it. This element is used in many areas of the app and can be super confusing for a VoiceOver user.
* The description is not read properly by VoiceOver when it’s multiline.
* It would be great if the no results title is the first element VoiceOver reads when a new screen is accessed. (Not sure if possible)

<p style="clear:both;"><sub>(internal reference: p77Llu-bQ6-p2 h/t @etoledom)</sub></p>

**Comment by designsimply** [comment]
Adding to Groundskeeping for review because it has a `[Pri] High` label.

**Comment by shiki** [comment]
> It would be great if the no results title is the first element VoiceOver reads when a new screen is accessed

I made a quick implementation of this. Please see [this video](https://drive.google.com/file/d/1-uekTEMabD8yv2HfxZn4KtrqZHOHDFbm/view?usp=sharing). I'm not actually sure if selecting anything (i.e. the “No likes yet“ message) right after the user selected a tab is desirable. It moves the focus away from the tabs. Then again, they'd probably move the focus to “see/hear” the results anyway. 🤔 

What do you think of this implementation, @designsimply and @etoledom? 

If you think this will work, what should happen if the list is not empty? Should we auto-select the first item in the list? Or something else? 🙂

**Comment by etoledom** [comment]
@shiki - It looks good!

A couple of questions: 
- Does it work without releasing the finger from the screen? (Usually the user will explore the whole screen with the finger without releasing).
- Will the focus go back to the tabs if the user continue dragging the finger around?

**Comment by shiki** [comment]
@etoledom 

> Does it work without releasing the finger from the screen? (Usually the user will explore the whole screen with the finger without releasing).

Unfortunately, no. The process looks like this:

1. Double tap on tab to activate
2. VoiceOver will focus on the empty list. 
3. The user will have to keep their finger over the last tap (step 1) and then touch and move around again. 

It seems difficult in terms of usage. What I'm more concerned about is when the user is navigating manually by swiping right and/or left. Since VoiceOver changes the focus, the user would most probably be confused about where they are. I tried to do this by closing my eyes and I was also confused. 😬 

Here is a diff if you'd like to try it out: https://gist.github.com/shiki/0f61636e9117b453b2bab8654ca3bbf8 

And here is a [video](https://drive.google.com/file/d/1-wmw9FNOZvxViz0a_Ct2s4rXJ2sY_uyl/view?usp=sharing) showing how it works. Starting from 0:34s, you'll see me navigating by swiping to the left to go back to the tab buttons. It takes a while.  

> Will the focus go back to the tabs if the user continue dragging the finger around?

Yes. But they just have to keep their finger over the last button so they don't lose track of where they were. It seems like a challenge. 

### Using Announcement

Your question gave me an idea of using `.announcement` instead. So I tried this: 

https://gist.github.com/shiki/1e7b71887288bdea433e46c5e63ca2fd

The idea was:

1. The user selects a tab 
2. If the list is empty, announce “No likes yet”

Unfortunately, it didn't work well because VoiceOver ignores announcements if it's already dictating something. It went like this instead:

1. The user selects the Likes tab.
2. VoiceOver dictates “Likes. Button”
3. The user double taps. This filters the notifications to likes only. 
4. VoiceOver dictates “Selected. Likes.”
3. App requests announcement “No likes yet”. VoiceOver ignored it since it was still dictating “Selected. Likes”

There are some instances where VoiceOver will successfully dictate “No likes yet”. But considering that it's not a reliable behavior, I think we'd rather not have it. 

I wonder if making the whole `NoResultsViewController` view an accessibility element, as suggested above, would be good enough. 🤔

**Comment by shiki** [comment]
> The root view should be a whole accessibility element to avoid VoiceOver from reading elements behind it.

I started implementing this for now (#12943).

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by kean** [comment]
Quick Start is removed in 25.3 – closing.
